src/test/sequenceTools.js

```diff
@@ -0,0 +1,681 @@
+import {defineSequence} from "../core/defineSequence";
```

Comment:
This is kind of approaching god-class territory, but I get why it's all in one file. Maybe split each function into its own file, then `export * as NonSlicingSequence from "./non-slicing-sequence" ...' etc?

Reply:
Yep yep this file was getting annoying to work with toward the end. I hadn't expected to need to add so many

This task is definitely on my radar, but also very low priority

Reply:
I created an issue: https://github.com/pineapplemachine/higher/issues/160